### PR TITLE
Pass along PGO type and use it in artifacts names

### PR DESCRIPTION
**Pull request by agocke**

### CODE REVIEW

File: eng/pipelines/runtime.yml [hunk 1]
```diff
@@ -151,15 +151,15 @@ jobs:
...
-#      pgoType: 'PGO'
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/coreclr/templates/build-job.yml
+    buildConfig: release
+    platforms:
+    - windows_x64
+    jobParameters:
+      testGroup: innerloop
+      pgoType: 'PGO'
```
Comment (safern) [inline eng/pipelines/runtime.yml:162]: do we expect different PGO types or why didn't you choose a bool?

Comment (agocke) [reply eng/pipelines/runtime.yml:162]: I couldn't figure out a good way in YAML to turn a bool into a string, and didn't want to repeat a conditional in the build where I change the name of jobs to include "PGO". If you have a better way I'd love to use it. I couldn't find any support for something like `if(pgoInstrument, 'PGO', '')` in YAML.

Comment (safern) [reply eng/pipelines/runtime.yml:162]: ah I see. Unfortunately there is no way in YAML to do that. So let's keep it as is.

**Comment by agocke** [comment]
@safern @jkoritzinsky Anything else?

**Review by safern — APPROVED** [review]

**Comment by safern** [comment]
@agocke if you haven't can you trigger an official build from this branch to make sure it is not broken?

**Comment by agocke** [comment]
@safern How do I trigger a pipeline on the branch of a fork? Wouldn't it have to be on the runtime repo itself?

**Comment by jkoritzinsky** [comment]
You can push your branch to the internal repo and manually trigger a build for your branch.

**Review by jkoritzinsky — APPROVED** [review]

**Comment by agocke** [comment]
Tested this in the official build and it's all green.

**Review by safern — APPROVED** [review]

**Pull request merged**